lib/bugsnag.rb

```diff
@@ -274,7 +298,28 @@ def check_endpoint_setup
         raise ArgumentError, "The session endpoint cannot be modified without the notify endpoint"
       end
     end
+
+    ##
+    # Convert the Report object to JSON
+    #
+    # We ensure the report is safe to send by removing recursion, fixing
+    # encoding errors and redacting metadata according to "meta_data_filters"
+    #
+    # @param report [Report]
+    # @return string
+    def report_to_json(report)
+      cleaner = Cleaner.new(
```

Comment:
Yeah; if it's cached then the `configuration.meta_data_filters` can get outdated if they're changed after the first time it's created

This could be solved by giving `Cleaner` access to the `Configuration` directly, but I'm not sure that's better

Reply:
After thinking about it more, it does make more sense to give it the `Configuration` object and construct it once. It then also cleans up the `rack_request` which currently makes its own cleaner which was slightly awkward with the new parameter, but after the change it can get the single shared instance instead